plans/internal/planproto/planfile.proto

```diff
@@ -17,6 +17,13 @@ message Plan {
     // the file for another message- or field-specific reason.
     uint64 version = 1;
 
+    // The mode that was active when this plan was created.
+    //
+    // This is saved only for UI purposes, so that Terraform can tailor its
+    // rendering of the plan depending on the mode. This must never be used to
+    // make decisions in Terraform Core during the applying of a plan.
+    Mode mode = 17;
```

Comment:
This is a good point, and the only thing that gives me pause is that we usually use "for display" to mean something than be _directly_ displayed to the user, rather than an indicator used as part of choosing what to display. But the string representations of these modes are not user-friendly, at least as currently designed.

I'm thinking about `ui_mode` as an alternative. I think it isn't _quite_ right but hopefully at least sufficient to prompt a potential user of it to notice the prefix is there and wonder what it means enough to come read this comment. What do you think?

Reply:
r.e. similar fields: I hesitate to generalize this too soon cause we only have two examples of this so far and they've been in different parts of the structure anyway. Since our plan file format isn't something we promise as stable between releases we will be free to reshape this later if we have new requirements that suggest a different design.

Reply:
Absolutely, I'm down with `ui_mode` - anything that makes the scope of that field unavoidably obvious to a developer (me!) is awesome.
I also agree (not that you said this directly) that my second suggestion is a premature optimization, and we can see what's actually useful in time.